category_encoders/hashing.py

```diff
@@ -238,18 +161,64 @@ def _transform_single_cpu(self, X, override_return_df=False):
         if not list(self.cols):
             return X
 
-        X = self.hashing_trick(X, hashing_method=self.hash_method, N=self.n_components, cols=self.cols)
-
-        if self.drop_invariant:
-            X = X.drop(columns=self.invariant_cols)
+        X = self.hashing_trick(
+            X, 
+            hashing_method=self.hash_method,
+            N=self.n_components, 
+            cols=self.cols, 
+        )
+        
+        return X
 
-        if self.return_df or override_return_df:
-            return X
-        else:
-            return X.to_numpy()
+    @staticmethod
+    def hash_chunk(shm_result, np_df, N, shm_offset):
+        for i, row in enumerate(np_df):
+            for val in row:
+                if val is not None:
+                    hasher = hashlib.md5()
+                    hasher.update(bytes(str(val), 'utf-8'))
+                    column_index = int.from_bytes(hasher.digest(), byteorder='big') % N
+                    row_index = (shm_offset + i)*N
+                    shm_index = row_index + column_index
+                    shm_result[shm_index] += 1
+
+    def hashing_trick_with_np_parallel(self, df, N):
+        np_df = df.to_numpy()
+        shm_result = multiprocessing.RawArray(HashingEncoder.default_int_np_array.dtype.char, len(df)*N)
+
+        n_process = []
+        chunk_size = int(len(np_df)/self.max_process)
+        ctx = multiprocessing.get_context(self.process_creation_method)
+        for i in range(0, self.max_process-1):
+            process = ctx.Process(target=self.hash_chunk,
+                args=(shm_result, np_df[i*chunk_size:((i+1)*chunk_size)], N, i*chunk_size))
+            n_process.append(process)
+
+        # The last process processes all the rest of the dataframe, because the number of rows might not
+        # be divisible by max_process. 
+        process = ctx.Process(target=self.hash_chunk,
+            args=(shm_result, np_df[(self.max_process-1)*chunk_size:], N, (self.max_process-1)*chunk_size))
+        n_process.append(process)
+
+        for process in n_process:
+            process.start()
+        for process in n_process:
+            process.join()
+
+        np_result = np.array(shm_result, 'int')
+
+        return pd.DataFrame(np_result.reshape(len(df), N), index=df.index)
 
     @staticmethod
-    def hashing_trick(X_in, hashing_method='md5', N=2, cols=None, make_copy=False):
+    def hashing_trick_with_np_no_parallel(df, N):
+        np_df = df.to_numpy()
+        np_result = np.zeros((len(df)*N), dtype='int')
+
+        HashingEncoder.hash_chunk(np_result, np_df, N, 0)
```

Comment:
do you need to re-assign this to `np_result` or will this be updated in place?

Reply:
It gets updated in place!